src/views/Home.jsx

```diff
@@ -1,12 +1,52 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createList } from '../api/firebase.js';
+import { useAuth } from '../api';
 
 export function Home({ data, setListPath }) {
+	const { user } = useAuth();
+	const userId = user?.uid;
+	const userEmail = user?.email;
+	const navigate = useNavigate();
+
+	const [listName, setListName] = useState('');
+
+	const handleInputChange = (event) => {
+		setListName(event.target.value);
+	};
+
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+		if (listName) {
+			const path = await createList(userId, userEmail, listName);
+			setListPath(path);
+			setListName('');
+			navigate('/list');
+			alert(`New list was created named: ${listName}`);
+		} else {
+			alert("Error occurred. List wasn't created");
```

Comment:
I think this error status is a little misleading given the data/logic you have present. You are checking `if (listName)`, which is a truthy conditional on a string, so this will really only return false if the string is empty `''`. I would just say that you could be more specific with your status about a listName not being present (which I'm not even sure is necessary given your `required` validation on the input), etc. 

So should this be a `try ... catch ...` around `createList` instead? Something to think about.